### PR TITLE
create group with permissions

### LINKED ISSUE #163: Command to Create Group for Each entry in UserGroup Enum class

### PR DESCRIPTION
**Pull request by ziadnabil**

### CODE REVIEW

**Review by mmostafa74 — APPROVED** [review]

**Review by mmostafa74 — CHANGES_REQUESTED** [review]
Add tests

**Comment by shahwan42** [comment]
https://adamj.eu/tech/2020/09/07/how-to-unit-test-a-django-management-command/

**Comment by mmostafa74** [comment]
Can you please rename file to create_group

File: xshop/core/management/commands/create_groups.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+class Command(BaseCommand):
+    def handle(self, *args, **options):
+
+        for group in GROUPS:
+            new_group, created = Group.objects.get_or_create(name=group)
+            for model in MODELS:
+                for permission in PERMISSIONS:
+                    name = "Can {} {}".format(permission, model)
+                    print("Creating {}".format(name))
```
Comment (shahwan42) [inline xshop/core/management/commands/create_groups.py]: use `self.stdout.write()` instead of print

**Comment by shahwan42** [comment]
@ziadnabil please try to finish this PR asap

**Comment by ziadnabil** [comment]
> @ziadnabil please try to finish this PR asap

still fixing permission problem :
Screenshot from 2021-05-03 16-07-13

File: xshop/core/management/commands/create_groups.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    "invoices": ["invoice"],
+    "orders": ["order", "orderitem"],
+    "products": ["product"],
+    "users": ["generalmanager", "manager", "dataentryclerk", "cashier"],
+    "shops": ["shop"],
+}
+ACTIONS = ["add", "change", "delete", "view"]
+
+
+class Command(BaseCommand):
```
Comment (mmostafa74) [inline xshop/core/management/commands/create_groups.py:14]: remove permissions, you are supposed to create groups only

**Review by mmostafa74 — CHANGES_REQUESTED** [review]

File: xshop/core/tests/test_views.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+from django.core.management import call_command
+from django.test import TestCase
+from django.contrib.auth.models import Group
+
+
+class CreateGroupsTests(TestCase):
+    def test_retrieve_existing_groups(self):
+        self.assertEqual(Group.objects.all().count(), 0)
+        call_command("create_groups")
+        self.assertEqual(Group.objects.all().count(), 5)
```
Comment (mmostafa74) [inline xshop/core/tests/test_views.py:11]: make sure that desired groups created

**Review by mmostafa74 — CHANGES_REQUESTED** [review]

File: xshop/core/tests/test_views.py (hunk 3 above)
Comment (mmostafa74) [reply xshop/core/tests/test_views.py:11]: self.assertTrue(Group.objects.get(name=UserGroup.CUSTOMER))

**Review by mmostafa74 — DISMISSED** [review]

**Pull request merged**